### PR TITLE
Add Event Types support

### PR DESCRIPTION
**Pull request by mthenw**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -369,6 +372,37 @@ class EGPlugin {
...
 
+  async configureEventTypes () {
+    let registeredEventTypes = await this.client.listEventTypes()
+
+    if (this.serverless.service.custom.eventTypes) {
+      const definedEventTypes = Object.keys(this.serverless.service.custom.eventTypes).map(name => {
+        return { name, authorizerId: this.serverless.service.custom.eventTypes[name].authorizerId }
+      })
+
+      await definedEventTypes.forEach(async eventType => {
```
Comment (raeesbhatti) [inline src/index.js]: Don't think `forEach` will work here [as expected](https://stackoverflow.com/a/37576787/2330318). You might need to consider changing this to a `for of`

File: src/index.js [hunk 2]
```diff
@@ -369,6 +372,37 @@ class EGPlugin {
...
+            registeredEventTypes = registeredEventTypes.filter(et => et.name !== eventType.name)
+          } else {
+            await this.client.updateEventType(eventType)
+            this.serverless.cli.consoleLog(`EventGateway: Event Type "${eventType}" updated.`)
+          }
+        }
+      })
+
+      // delete unused event types
+      await registeredEventTypes.forEach(async eventType => {
```
Comment (raeesbhatti) [inline src/index.js]: Same here

File: src/index.js (hunk 1 above)
Comment (mthenw) [reply src/index.js]: Fixed, thanks for the info.

**Comment by mthenw** [comment]
@raeesbhatti will do. Thanks for the tip!

**Pull request merged**